src/main/java/de/zalando/aruha/nakadi/service/subscription/state/ClosingState.java

```diff
@@ -50,6 +50,9 @@ public void onEnter() {
     }
 
     private void onTopologyChanged() {
+        if (topologyListener == null) {
+            throw new IllegalStateException("topologyListener should not be null when calling onTopologyChanged method");
```

Comment:
Ok, I see. That might happen because there is an scheduled task which could be executed at any time, inclusive after `onExit` which assigns `null` to `topologyListener`.

Failing during listener creation would not cover all cases, so this solution is probably more appropriate.